core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -1733,10 +1733,15 @@ struct ParallelReduceFence {
 // ReturnValue is scalar or array: take by reference
 
 template <class PolicyType, class FunctorType, class ReturnType>
-inline typename std::enable_if<
-    Kokkos::is_execution_policy<PolicyType>::value>::type
+inline std::enable_if_t<Kokkos::is_execution_policy<PolicyType>::value &&
+                        !(Kokkos::is_view<ReturnType>::value ||
+                          Kokkos::is_reducer<ReturnType>::value)>
 parallel_reduce(const std::string& label, const PolicyType& policy,
                 const FunctorType& functor, ReturnType& return_value) {
+  static_assert(
+      !std::is_const<ReturnType>::value,
+      "A const return type is only allowed for a View or reducer return type!");
```

Comment:
Did you check in our Wiki how we actually refer to the reducer argument?  I feel like "return type" might be confusing

Reply:
I changed it to "reduction result type".